modules/io/save.py

```
@@ -0,0 +1,68 @@
+    def __init__(
+        self,
+        router: AkariRouter,
+        logger: AkariLogger,
+    ) -> None:
+        super().__init__(router, logger)
+
+    def call(self, data: AkariData, params: _SaveModuleParams, callback: AkariModuleType | None = None) -> AkariDataSet:
+        try:
+            save_data = data.last().__dict__[params.save_from_data]
+        except KeyError:
+            raise ValueError(f"Data does not contain the key '{params.save_from_data}'.")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)